src/wayland/shell/wlr_layer/handlers.rs

```diff
@@ -308,18 +308,21 @@ where
     }
 
     fn destroyed(
-        _state: &mut D,
+        state: &mut D,
         _client_id: wayland_server::backend::ClientId,
         object_id: wayland_server::backend::ObjectId,
         data: &WlrLayerSurfaceUserData,
     ) {
         data.alive_tracker.destroy_notify();
+
         // remove this surface from the known ones (as well as any leftover dead surface)
         data.shell_data
             .known_layers
             .lock()
             .unwrap()
             .retain(|other| other.shell_surface.id() != object_id);
+
+        WlrLayerShellHandler::destroyed(state);
```

Comment:
Well I think it would be helpful to send the object removed during retain to the user (perhaps by replacing it with remove(), as there is always just one surface with said objectid) 

And same for xdg as well.